.github/workflows/linux_ci.yml

```diff
@@ -0,0 +1,54 @@
+name: Run Tests
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  tests:
+    name: Ubuntu Tests
+    env:
+      ENV_NAME: hera_qm_tests
+      PYTHON: ${{ matrix.python-version }}
+      OS: ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [3.6, 3.7, 3.8]
+
+    steps:
+    - uses: actions/checkout@master
+      with:
+        fetch-depth: 1
+
+    - name: Get Miniconda
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O $HOME/miniconda.sh;
+        bash $HOME/miniconda.sh -b -p $HOME/miniconda
+
+    - name: Setup Environment
+      run: |
+        export PATH="$HOME/miniconda/bin:$PATH"
+        ./ci/install_conda.sh
+
+
+    - name: Install
+      run: |
+        export PATH="$HOME/miniconda/bin:$PATH"
+        source activate ${ENV_NAME}
+        pip install .
+
+    - name: Run Tests
+      run: |
+        export PATH="$HOME/miniconda/bin:$PATH"
+        source activate $ENV_NAME
+        pytest --cov-report xml:./coverage.xml
+        pycodestyle
+        pydocstyle hera_qm/*.py --convention=numpy
```

Comment:
While there's no reason you can't leave these as one giant part of an Action but you can also split them up into separate jobs entirely if you wan. Either way up to you!